plugins/infinispan52/src/main/java/org/radargun/service/InfinispanHotrodService.java

```diff
@@ -97,8 +100,8 @@ protected TcpTransportFactory getTransportFactory(RemoteCacheManager manager) {
    @Override
    public Map<String, Number> getValues() {
       Map<String, Number> values = new HashMap<>();
-      TcpTransportFactory nrFactory = getTransportFactory(managerNoReturn);
-      TcpTransportFactory frFactory = getTransportFactory(managerForceReturn);
+      TcpTransportFactory nrFactory = (TcpTransportFactory) getTransportFactory(managerNoReturn);
+      TcpTransportFactory frFactory = (TcpTransportFactory) getTransportFactory(managerForceReturn);
```

Comment:
Why in this context a class not found exception will not be thrown?

Reply:
Because of https://github.com/radargun/radargun/blob/master/core/src/main/java/org/radargun/config/InitHelper.java#L86

Reply:
Just a nit, you can cast on this linehttps://github.com/radargun/radargun/blob/62f349c805206ff7ea35cf296e5934814363e860/plugins/infinispan52/src/main/java/org/radargun/service/InfinispanHotrodService.java#L84 and keep the return as TcpTransportFactory instead of Object, on this way you can avoid cast every time you call this method like https://github.com/radargun/radargun/blob/62f349c805206ff7ea35cf296e5934814363e860/plugins/infinispan52/src/main/java/org/radargun/service/InfinispanHotrodService.java#L103

Reply:
If a method is using a class that is not present in the plugin classpath `infinispan-100` a class not found will throw. The method is recursive and will check all [getSuperclass](https://github.com/radargun/radargun/blob/master/core/src/main/java/org/radargun/config/InitHelper.java#L102)